src/lib/field/_selector.scss

```diff
@@ -545,11 +546,27 @@
         @include base.icon-color(invalid);
       }
     }
-    &--disabled {
+    &--disabled:not(.forge-field__addon-end-container--enabled) {
       ::slotted([slot='addon-end']) {
         @include base.icon-color(disabled);
       }
     }
   }
+
+  // Addon End Slot Always Enabled.
+  &.forge-field__addon-end-container--enabled.forge-field--disabled:not(.forge-field--invalid) {
+    .forge-field__addon-end-container {
+      @include theme.property(background-color, surface);
```

Comment:
While I understand that we need this background override due to our current DOM structure, it does go against the original intent of the component to inherit the container background when enabled. Not a big deal, and someone could override per their needs if they run into it.

Just wanted to call it out so the expectation is clear, and so we can discuss when we look at this for v3.

Reply:
@DRiFTy17 just to clarify the DOM positioning of this doesn't allow us to use transparent as the color here as the background color behind the add on would show through, which is the disabled. I agree the layering isn't ideal but this seems like the correct solution for the moment. Definitely worth a discussion for v3

Reply:
Are you just documenting the intent here @DRiFTy17 , or is there something that you would like changed about this?